driver-core/src/main/com/mongodb/DuplicateKeyException.java

```diff
@@ -19,9 +19,21 @@
 import org.bson.BsonDocument;
 
 /**
- * Subclass of {@link WriteConcernException} representing a duplicate key exception
+ * A subclass of {@link WriteConcernException} representing a duplicate key exception
+ *
+ * <p>
+ * This class is effectively deprecated. It is only thrown with use of the long-deprecated MongoClient.getDB method and associated
```

Comment:
What is the reason we don't want to use `@Deprecated` and `@deprecated` here and in `WriteConcernException.java`?

Reply:
The Javadoc for `MongoClient#getDB` states:

>    The {@link DB} class has been superseded by {@link com.mongodb.client.MongoDatabase}.  The deprecation of this method effectively deprecates the {@link DB}, {@link DBCollection}, and {@link DBCursor} classes, among others; but in order to give users time to migrate to the new API without experiencing a huge number of compiler warnings, those classes have not yet been formally deprecated.

These two exception classes would be classified as part of the `among others` phrase in that documentation.

Reply:
I see, thanks you. For the record, I would like to express that I disagree with the approach of deprecating code elements in such a way that tools cannot spot the deprecation.